### PR TITLE
feat(Storybook): Initial setup of how storybook can be used

### LINKED ISSUE #228: Storybook usage for component development
Storybook allows you to test components separately from the application itself. Lets say you want to implement a feature that shows a better screen if no results are returned for a table, instead of changing the URL or just setting the data to empty instantly, you can use a Storybook and provide this information to a component directly instead

We could possibly use this for [unit tests too](https://storybook.js.org/docs/react/writing-tests/introduction)

https://storybook.js.org/

### PR DESCRIPTION
**Pull request by SpyTec**
Basic concept of what storybooks are. I've not added a makefile yet so to run go to frontend/ and run `yarn storybook`

If you are in the root repository, you can do `make storybook`. I might change this to dev-storybook or something.

TODO:
- [x] Flesh out the stories a bit more
- [x] Put test data outside the stories components

Fixes #228 

Related to

Merge after
#246 

---

RELEASE NOTES BEGIN
Setup Storybook for component development and showcase
RELEASE NOTES END

### CODE REVIEW

**Comment by SpyTec** [comment]
image

image

image

**Review by lachmanfrantisek — COMMENTED** [review]
I really like the outcome, but sadly don't understand the code much...;( (But that's mainly an issue on my side...;) So, I am :+1:  for going this way.

Can you please describe the approach a bit in the [contribution guide](https://github.com/packit/dashboard/blob/main/CONTRIBUTING.md)?

File: frontend/public/mockServiceWorker.js
Comment (SpyTec) [inline frontend/public/mockServiceWorker.js:1]: Will omit this from VCS
https://github.com/mswjs/msw/discussions/1015

**Comment by SpyTec** [comment]
@lachmanfrantisek I updated the `CONTRIBUTING.md` file. WDYT?

**Comment by SpyTec** [comment]
Current issue, `preview.ts` seems to be parsed as a `.js` file as it complains about `const preview: Preview =`, which is fixed if `Preview` is removed
image

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -74,6 +74,95 @@ echo fs.inotify.max_user_watches=524288 | sudo tee -a /etc/sysctl.conf && sudo s
...
+const Template = (args) => <NotFound {...args} />;
+
+// Next we call bind, this allows us to use `this` properly within the component
+// https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Function/bind
+export const Default = Template.bind({});
+```
+
+#### Network usage
+
+Most of our componments need to use some sort of network activity, likely an API call to Packit service. For those stories, we are using a library called [MSJ](https://mswjs.io/).
```
Comment (jpopelka) [inline CONTRIBUTING.md:123]:
```suggestion
Most of our components need to use some sort of network activity, likely an API call to Packit service. For those stories, we are using a library called [MSJ](https://mswjs.io/).
```

**Review by jpopelka — COMMENTED** [review]
some module error prevented me from trying it, but :+1:

**Comment by SpyTec** [comment]
@jpopelka should be working now. Want to give it a go? Need to redo the installation of node modules

File: CONTRIBUTING.md (hunk 1 above)
Comment (SpyTec) [reply CONTRIBUTING.md]: Fixed, thanks

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -74,6 +74,95 @@ echo fs.inotify.max_user_watches=524288 | sudo tee -a /etc/sysctl.conf && sudo s
 
 [Technical details](https://github.com/guard/listen/wiki/Increasing-the-amount-of-inotify-watchers#the-technical-details)
 
+## Developing frontend with Storybook
+
+For the dashboard's frontend, we make use of Storybook to test out different scenarios that each component can be in. For example, testing that everything looks fine for accessibility point-of-view, or that network errors displays correctly
+
+Storybook allows us to isolate individual React components and display them in different scenarios that we can view with minimal interferance. Allowing us to do a proper integration test
+
+However, Storybook itself is not a replacement for making sure the code works correctly. The API endpoints being used might change in the future, which is why it is good to test your PRs out against stage before production deploymnent
```
Comment (jpopelka) [inline CONTRIBUTING.md:83]:
```suggestion
For the dashboard's frontend, we make use of Storybook to test out different scenarios that each component can be in. For example, testing that everything looks fine from an accessibility point-of-view, or that network errors are displayed correctly.

Storybook allows us to isolate individual React components and display them in different scenarios that we can view with minimal interference. Allowing us to do a proper integration test.

However, Storybook itself is not a replacement for making sure the code works correctly. The API endpoints being used might change in the future, which is why it is good to test your PRs out against stage before production deployment.
```

File: CONTRIBUTING.md [hunk 3]
```diff
@@ -74,6 +74,95 @@ echo fs.inotify.max_user_watches=524288 | sudo tee -a /etc/sysctl.conf && sudo s
...
+const Template = (args) => <NotFound {...args} />;
+
+// Next we call bind, this allows us to use `this` properly within the component
+// https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Function/bind
+export const Default = Template.bind({});
+```
+
+#### Network usage
+
+Most of our components need to use some sort of network activity, likely an API call to Packit service. For those stories, we are using a library called [MSJ](https://mswjs.io/).
```
Comment (jpopelka) [inline CONTRIBUTING.md:123]:
```suggestion
Most of our components need to use some sort of network activity, likely an API call to Packit service. For those stories, we are using a library called [MSW](https://mswjs.io/).
```

**Review by jpopelka — COMMENTED** [review]
Works OK now.

Are the `unmet peer dependency` warnings expected when running `yarn install`?

**Comment by SpyTec** [comment]
I'd say we're ready to merge this, can improve later

There is one thing I want answered so asked over in a plugin repo, but good for now
https://github.com/JesusTheHun/storybook-addon-react-router-v6/discussions/37

**Pull request merged**